esrally/client/factory.py

```diff
@@ -222,22 +224,28 @@ async def on_request_end(session, trace_config_ctx, params):
 
         # override the builtin JSON serializer
         self.client_options["serializer"] = LazyJSONSerializer()
-        # TODO see esrally.client/asynchronous
-        # self.client_options["trace_config"] = trace_config
 
         if api_key is not None:
             self.client_options.pop("http_auth")
             self.client_options["api_key"] = api_key
 
-        return RallyAsyncElasticsearch(
+        async_client = RallyAsyncElasticsearch(
             distro=self.distribution_version,
             hosts=self.hosts,
             transport_class=RallyAsyncTransport,
             ssl_context=self.ssl_context,
-            maxsize=max,
+            maxsize=self.max_connections,
             **self.client_options,
         )
 
+        # the AsyncElasticsearch constructor automatically creates the corresponding NodeConfig objects, so we set
+        # their instance attributes after they've been instantiated
+        for node_connection in async_client.transport.node_pool.all():
+            node_connection.trace_configs = trace_config
```

Comment:
I would argue for `node_connection.trace_configs = [trace_config]` here.

Reply:
Addressed via https://github.com/elastic/rally/pull/1669/commits/c85a5ace7ddce84ae56def87beafff6361875138